sops.go

```diff
@@ -99,6 +100,9 @@ func (tree TreeBranch) walkValue(in interface{}, path []string, onLeaves func(in
 
 func (tree TreeBranch) walkSlice(in []interface{}, path []string, onLeaves func(in interface{}, path []string) (interface{}, error)) ([]interface{}, error) {
 	for i, v := range in {
+		if _, ok := v.(yaml.Comment); ok {
+			continue
+		}
```

Comment:
Would there be a way to handle this that doesn't break the boundary between sops and the data type? In other words: can sops be unaware of the `yaml.Comment` type?